### PR TITLE
Feature/syncronous producer

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Refactored splitter to use the synchronous Kafka producer instead of the async producer. This makes it a lot simpler to handler errors and track the progress through the file. 

### How to review

Test the CSV splitter and ensure it can process files in a reasonable time.

### Who can review

Anyone

### CODE REVIEW

File: config/splitter.go [hunk 1]
```diff
@@ -47,7 +51,12 @@ func init() {
 		TopicName = topicNameEnv
 	}
 
-
+	batchSizeEnv, err := strconv.ParseInt(os.Getenv(batchSizeKey), 10, 16);
```
Comment (githubmo) [inline config/splitter.go]: So our batch size can't be bigger than 65536 messages? Fair assumption.

File: config/splitter.go [hunk 2]
```diff
@@ -47,7 +51,12 @@ func init() {
...
+	if err != nil {
+		log.Error(err, log.Data{"message":"Failed to parse batch size. Using default." })
+	} else {
+		BatchSize = int(batchSizeEnv)
```
Comment (githubmo) [inline config/splitter.go]: There's something a bit unsettling for me that we take an `int16`, put it into an `int64` and then cast it to `int32`.

Comment (githubmo) [reply config/splitter.go]: Could we not have avoided all of this boxing and unboxing if we just did `batchSizeEnv, err := strconv.Atoi(v)` above?

https://golang.org/pkg/strconv/#Atoi

Comment (githubmo) [reply config/splitter.go]: I feel there is a hidden `panic` here we may not be accounting for.

File: splitter/processor.go [hunk 3]
```diff
@@ -39,55 +39,75 @@ func createMessage(index int, row []string) Message {
...
+	var batchNumber = 1
+	var isFinalBatch = false
+
+batchLoop:
+	for { // each batch
+
+		log.Debug("Processing batch number "+strconv.Itoa(batchNumber)+" index: "+strconv.Itoa(index), nil)
+		var msgs []*sarama.ProducerMessage = make([]*sarama.ProducerMessage, batchSize)
+
+	createBatchLoop:
```
Comment (githubmo) [inline splitter/processor.go]: Instead of using `break createBatchLoop`, we can remove the need for `createBatchLoop` and wrap the code in an `if ... else` clause.

Comment (githubmo) [reply splitter/processor.go]: Also, we can add the `isFinalBatch` to that clause so as not to have `batchLoop`.

**Review by githubmo — CHANGES_REQUESTED** [review]
Overall LGTM, code is mostly neat and I've learnt a couple of good tricks here and there. If you can implement the changes then I can merge this change as soon as possible.

File: config/splitter.go (hunk 2 above)
Comment (CarlHembrough) [reply config/splitter.go]: I have updated the code to use `strconv.Atoi(v)` as suggested

File: splitter/processor.go [hunk 4]
```diff
@@ -39,55 +39,75 @@ func createMessage(index int, row []string) Message {
...
 				if err == io.EOF {
 					log.Debug("EOF reached, no more records to process", nil)
-					break csvLoop
+					isFinalBatch = true
+					msgs = msgs[0:batchIndex] // the last batch is smaller than batch size, so resize the slice.
+					log.Debug(strconv.Itoa(batchIndex)+" messages in the final batch.", nil)
+					break createBatchLoop
 				} else {
 					fmt.Println("Error occored and cannot process anymore entry", err.Error())
 					panic(err)
```
Comment (ian-kent) [inline splitter/processor.go:63]: i know its not part of the PR, but a panic seems extreme since it'll kill the entire app

File: splitter/processor.go (hunk 3 above)
Comment (CarlHembrough) [reply splitter/processor.go]: I have refactored out the labelled break statements.

**Review by githubmo — APPROVED** [review]
